tests/test_modules.py

```diff
@@ -776,3 +805,36 @@ def forward(self, x):
         self.assertEqual(cnt.op_count, 1)
         self.assertTrue(torchdynamo.testing.same(pre, opt_pre))
         self.assertTrue(torchdynamo.testing.same(out1, out_post))
+
+    def test_lazy_module(self):
+        input_shape = (16, 3, 6, 7, 8)
+        cnt = torchdynamo.testing.CompileCounter()
+        with torchdynamo.optimize(cnt):
+
+            def test_dynamic_module():
+                module = LazyModule()
+                input = torch.ones(*input_shape)
+                module(input)
```

Comment:
we should call this multiple times and assert cnt.frame_count is 1 (to check for guards failing)